cmd/gitopsUpdateDeployment.go

```diff
@@ -0,0 +1,158 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/docker"
+	gitUtil "github.com/SAP/jenkins-library/pkg/git"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/go-git/go-git/v5/plumbing"
+	"io"
+	"path/filepath"
+)
+
+type gitopsGitUtils interface {
+	CommitSingleFile(filePath, commitMessage string, worktree gitUtil.UtilsWorkTree) (plumbing.Hash, error)
+	PushChangesToRepository(username, password string, repository gitUtil.UtilsRepository) error
+	PlainClone(username, password, serverURL, directory string) (gitUtil.UtilsRepository, error)
+	ChangeBranch(branchName string, worktree gitUtil.UtilsWorkTree) error
+	GetWorktree(repository gitUtil.UtilsRepository) (gitUtil.UtilsWorkTree, error)
+}
+
+type gitopsFileUtils interface {
+	TempDir(dir, pattern string) (name string, err error)
+	RemoveAll(path string) error
+}
+
+type gitopsExecRunner interface {
+	RunExecutable(executable string, params ...string) error
+	Stdout(out io.Writer)
+	Stderr(err io.Writer)
+}
+
+var gitUtilities gitopsGitUtils = gitUtil.TheGitUtils{}
+var fileUtilities gitopsFileUtils = piperutils.Files{}
+
+func gitopsUpdateDeployment(config gitopsUpdateDeploymentOptions, telemetryData *telemetry.CustomData) {
+	// for command execution use Command
+	var c gitopsExecRunner = &command.Command{}
+	// reroute command output to logging framework
+	c.Stdout(log.Writer())
+	c.Stderr(log.Writer())
+
+	// for http calls import  piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	// and use a  &piperhttp.Client{} in a custom system
+	// Example: step checkmarxExecuteScan.go
+
+	// error situations should stop execution through log.Entry().Fatal() call which leads to an os.Exit(1) in the end
+	err := runGitopsUpdateDeployment(&config, c)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runGitopsUpdateDeployment(config *gitopsUpdateDeploymentOptions, command gitopsExecRunner) error {
+	temporaryFolder, tempDirError := fileUtilities.TempDir(".", "temp-")
```

Comment:
no need to add specific error variables.

`err` will do here and in all further places.